tests/testthat/test-cell_suppression.R

```diff
@@ -51,6 +51,69 @@ test_that("determine_cell_suppression correctly suppresses cells", {
   testthat::expect_equal(soln, test$suppress, check.attributes = FALSE)
 })
 
+test_that("Correctly deals with constraints hidden by prior cells", {
+  # A prior implementation did not conduct a final pass over cells,
+  # resulting in one cell implementing a later constraint that interfered
+  # with a prior set of constraints that passed initial solutions before
+  # all cells had been assessed
+  n <- c(262L, 78L, 340L, 110L, 59L, 169L, 52L, 16L, 68L, 54L, 16L, 70L, 97L, 17L,
+         114L, 90L, 37L, 127L, 1L, 3L, 4L, 0L, 0L, 0L, 666L, 226L, 892L)
+  ns <- matrix(c(1, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0,
+                 1, 0, -1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 1, 0, 0, 0, 0, 0,
+                 0, 1, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, -1, 0, 0, 0,
+                 0, 0, 0, 0, 0, 1, 0, 0, 1, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 1, 0,
+                 0, 0, 0, 0, 0, 1, 0, 0, 0, -1, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0,
+                 0, 1, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 1, 0, 0,
+                 0, 0, -1, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 1, 0, 0, 0, 1, 0,
+                 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, -1, 0, 0, 0,
+                 0, 0, 1, 0, 0, 0, 0, 0, 1, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 1, 0,
+                 0, 0, 1, 0, 0, 0, 0, 0, 0, -1, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0,
+                 0, 1, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 1, 0, 0, 0, 0, 0,
+                 0, 0, -1, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 1, 1, 0, 0, 0, 0,
+                 0, 0, 0, 0, 0, 1, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, -1, 0, 0, 1,
+                 1, 1, 1, 1, 1, 1, 1, 1, -1, 0, 0, 1, 1, 1, 1, 1, 1, 1, 1, 0,
+                 -1, 0, -1, -1, -1, -1, -1, -1, -1, -1, 0, 0, -1), ncol = 27)
+
+  soln <- c(FALSE, FALSE, FALSE, FALSE, FALSE, FALSE, TRUE, TRUE, TRUE,
+            FALSE, FALSE, FALSE, FALSE, FALSE, FALSE, FALSE, FALSE, FALSE,
+            TRUE, TRUE, TRUE, FALSE, FALSE, FALSE, FALSE, FALSE, FALSE)
+
+  expect_identical(determine_cell_suppression(n, ns), soln)
+})
+
+test_that("Solves highly dimensional tables that may return non-integer LP solutions", {
```

Comment:
I think we want this test, but I’d prefer if we didn’t have to trust that MultiMappingTable’s ordering won’t change in future. It’s also large enough that we probably don’t want to have the matrix in code.

How about saving the expected suppression/test counts/test nullspace to a (say) tab-separated file and writing a test helper (say in tests/testthat/helper-supressions.R) to read those three things in (and maybe even write those things out). Then, it would be a touch easier to read and verify the test data.

Suggested tab file format something like desired `suppression` in first column, test `count` in second column and then rows of the nullspace in subsequent columns `NS1`, `NS2`, etc. So effectively we would write out `cbind(suppression = suppress, counts = n, NS = t(nullspace))` with something that handles matrix-columns.

Reply:
That's a good idea! I have converted these two into .tsv files and added a helper function as you suggested. Tests passing